### PR TITLE
Watch the image pull secret and propagate it to components

### PR DESCRIPTION
**Pull request by yanmxa**
Signed-off-by: myan <myan@redhat.com>
Ref: https://issues.redhat.com/browse/ACM-4732

### CODE REVIEW

**Comment by yanmxa** [comment]
override "SonarCloud Code Analysis"

File: operator/pkg/controllers/hubofhubs/globalhub_config.go [hunk 1]
```diff
@@ -18,6 +18,8 @@ func (r *MulticlusterGlobalHubReconciler) reconcileSystemConfig(ctx context.Cont
 		return err
 	}
 
+	config.SetImagePullSecretName(mgh)
```
Comment (morvencao) [inline operator/pkg/controllers/hubofhubs/globalhub_config.go:21]: can we do this in predicate function to avoid the race condition and missing call?

Comment (yanmxa) [reply operator/pkg/controllers/hubofhubs/globalhub_config.go:21]: @morvencao Thanks for your review! The config reconciling will always be in front of the manager and Grafana reconciles (which will use the image pull secret name). So I think the race condition may not exist in such case. What do you think of it?

image

Comment (morvencao) [reply operator/pkg/controllers/hubofhubs/globalhub_config.go:21]: it makes sense.

**Comment by morvencao** [comment]
leave this to @clyang82 and @ldpliu for future review.

**Pull request merged**